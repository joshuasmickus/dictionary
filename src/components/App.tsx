import * as React from 'react';
import {
  BrowserRouter as Router,
  Link,
  Route
} from 'react-router-dom';

import CreateDictionary from './CreateDictionary';
import DictionaryList from './DictionaryList';

export const App: React.FC = () => (
  <Router>
    <h1>Dictionary</h1>
    <Link to="/">Home</Link>&nbsp;
    <Link to="/create">Create Dictionary</Link>
    <Route path="/" exact={true} component={DictionaryList} />
    <Route path="/create" component={CreateDictionary} />
  </Router>
);
