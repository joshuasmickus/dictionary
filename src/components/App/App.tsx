import * as React from 'react';
import {
  BrowserRouter as Router,
  Link,
  Route
} from 'react-router-dom';
import { ToastContainer } from 'react-toastify';

import CreateDictionary from '../CreateDictionary';
import CreateDictionaryItem from '../CreateDictionaryItem';
import DictionaryItemList from '../DictionaryItemList';
import DictionaryList from '../DictionaryList';
// import EditDictionaryItem from '../EditDictionaryItem';

export const App: React.FC = () => (
  <Router>
    <h1>Dictionary</h1>
    <Link to="/">Home</Link>&nbsp;
    <Link to="/create-dictionary">Create Dictionary</Link>
    <Route path="/" exact={true} component={DictionaryList} />
    <Route path="/create-dictionary" component={CreateDictionary} />
    <Route path="/edit-dictionary/:id" component={DictionaryItemList} exact={false} />
    <Route path="/edit-dictionary/:id/add-item" component={CreateDictionaryItem} exact={true} />
    {/* <Route path="/edit-dictionary/:id/edit-item/:itemId" component={EditDictionaryItem} exact={true} /> */}
    <ToastContainer />
  </Router>
);
