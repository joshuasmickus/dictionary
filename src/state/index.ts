import { combineReducers, compose, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { persistReducer, persistStore } from 'redux-persist';
import storage from 'redux-persist/lib/storage';

import { dictionaryReducer } from './dictionary';

export const rootReducer = combineReducers({
  dictionary: dictionaryReducer
});

const enhance = compose(
  composeWithDevTools()
);

const persistConfig = {
  key: 'root',
  storage,
}

const persistedReducer = persistReducer(persistConfig, rootReducer)

export function configureStore() {
  const store = createStore(persistedReducer, enhance);
  const persistor = persistStore(store);

  return { store, persistor };
}
