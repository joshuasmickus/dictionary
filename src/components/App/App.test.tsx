import * as React from 'react';
import { Provider } from 'react-redux';
import { render } from 'react-testing-library';


import { configureStore } from '../../state';
import { App } from './App';

const { store } = configureStore();

describe('Given a App component', () => {
  describe('when it is rendered', () => {
    it('should match the snapshot', () => {
      const { container } = render(<Provider store={store}><App /></Provider>);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
