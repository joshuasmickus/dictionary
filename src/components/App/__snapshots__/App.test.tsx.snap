// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a App component when it is rendered should match the snapshot 1`] = `
<h1>
  Dictionary
</h1>
`;
